### ISSUE TITLE
Dependent Dropdowns

### ISSUE THREAD

**Issue by rajivseelam** [trebuchetty/Handsontable-select2-editor#15]
This is a follow up to:

https://github.com/trebuchetty/Handsontable-select2-editor/issues/14

The solution used is:

    {
      data: 'country_id',
      editor: 'select2', 
      renderer: customDropdownRenderer,
      select2Options: { 
              data: {!! $production_units !!} ,
              dropdownAutoWidth: true,
      }
    },
    {
      data: 'state_id',
      editor: 'select2', 
      renderer: customDropdownRenderer,
      select2Options: { 
              data: [],
              dropdownAutoWidth: true,
              width: 'resolve'
      }
    },

    afterChange: function (change, source) {

    if(change)
    {

      if(change[0][1] == 'country_id')
      {
        $.get("/api/states?country="+change[0][3], function(data){

             hot.getCellMeta(row,1).select2Options.data = data;

        });
      }

    }

  },

Let's say I chose "Australia" in first row, it shows states in Australia in the states column of the corresponding row.

In another row below it, If i chose "India", then the problem starts, The states data of first row in which we selected "Australia" is also changes to states list of "India".

In short, select2Options are changing for all rows.

I guess it only keeps one instance per column, so data is common across all columns.

Is there anyway to solve this issue?

**Comment by trebuchetty** [comment]
I'm not sure how to solve this one. Handsontable keeps the one instance alive per column. You might need to override this behavior, which I think is possible, but there would be performance penalty associated with that.
You might need to validate the other column's value before opening the editor so that you could replace the selection list with the correct list.

**Comment by rajivseelam** [comment]
Requesting to re-open the issue.

An example is hosted at jsfiddle as a follow-up to comment by handsontable: https://twitter.com/handsontable/status/654946829939920896

http://jsfiddle.net/jepod675/2/

![jsfiddle-select2](https://cloud.githubusercontent.com/assets/1103649/11436121/7df4e5ce-9507-11e5-92bd-2da42365a553.png)
